moped-editor/src/views/dev/LookupsView/index.js

```diff
@@ -31,6 +40,52 @@ const LookupsView = () => {
     fetchPolicy: "no-cache",
   });
 
+  const [selectedRecordKey, setSelectedRecordKey] = useState(null);
+
+  /**
+   * We're using history here (and elsewhere) because it's not possible to use react-router
+   * to replace window.loctation w/o forcing a re-render.
+   * See // https://github.com/remix-run/react-router/issues/8908
+   */
+  const history = createBrowserHistory();
+
+  /**
+   * Create a ref index for every table element on the page so that we can scroll to them
+   * */
+  const refs = useMemo(
+    () =>
+      SETTINGS.reduce((prev, recordType) => {
+        prev[recordType.key] = createRef();
+        return prev;
+      }, {}),
+    []
+  );
+
+  /**
+   * Use the record hash from the URL, if present. This only happens on page load.
+   * */
+  const { hash: recordKeyHash } = useLocation();
+  useEffect(() => {
+    if (!recordKeyHash) {
+      return;
+    }
+    const recordKey = recordKeyHash.replace("#", "").replace("-", "_");
+    setSelectedRecordKey(recordKey);
+  }, [recordKeyHash]);
+
+  /**
+   * Listens for changes to the selected record key and scrolls to its table
+   */
+  useEffect(() => {
+    if (!selectedRecordKey) {
+      return;
+    }
+    const ref = refs?.[selectedRecordKey];
+    if (ref?.current) {
+      ref.current.scrollIntoView({ behavior: "smooth" });
```

Comment:
smooth scrolling isn't everyone's cup of tea—happy to change it to `auto`.

Reply:
i dig it. nice touch 💅